core/src/test/java/cucumber/runtime/formatter/PrettyFormatterTest.java

```diff
@@ -381,6 +382,37 @@ public void should_mark_arguments_in_steps() throws Throwable {
                                           AnsiEscapes.GREEN + AnsiEscapes.INTENSITY_BOLD + "arg2"  + AnsiEscapes.RESET));
     }
 
+    // Reproduce issue #619: This test should now fail!
+    @Test(expected = StringIndexOutOfBoundsException.class)
+    public void nested_capture_groups_cannot_be_handled(){
+        Formats formats = new AnsiFormats();
+        Argument fullArg = new Argument(0, "this is the nested argument full argument");
+        Argument nestedArg = new Argument(-29, "nested argument");
+        PrettyFormatter prettyFormatter = new PrettyFormatter(null);
+
+        try {
+            String formattedText = prettyFormatter.formatStepText("Given ", "this is the (nested argument) full argument", formats.get("passed"), formats.get("passed_arg"), asList(fullArg, nestedArg));
+        } catch (StringIndexOutOfBoundsException e) {
+            assertEquals("String index out of range: -70", e.getMessage());
+            throw e;
+        }
+    }
+
+    @Test
+    public void nested_capture_groups_cannot_be_formatted(){
+        Formats formats = new AnsiFormats();
+        Argument fullArg = new Argument(0, "this is the nested argument full argument");
+        Argument nestedArg = new Argument(-29, "nested argument");
+        PrettyFormatter prettyFormatter = new PrettyFormatter(null);
+
+        try {
+            String formattedText = prettyFormatter.formatStepText("Given ", "this is the (nested argument) full argument", formats.get("passed"), formats.get("passed_arg"), asList(fullArg, nestedArg));
+        } catch (StringIndexOutOfBoundsException e) {
+            assertEquals("String index out of range: -70", e.getMessage());
```

Comment:
Right now this test is asserting that an exception which it doesn't expect to be thrown has the right format. I would expect a test that checks if the formatted text is properly formatted. For example see: `should_mark_arguments_in_steps`